### DISCUSSION TITLE
findByCursor - orderBy with Date results in getCollection where-clause without new Date()

### DISCUSSION THREAD

**Discussion by lod911** [mikro-orm/mikro-orm#5478] [General]
### Describe the bug

Hello

With mongodb-driver the findByCursor function doesn't respect the mongodb-syntax in the where clause (from the orderBy) for Date-Attributes.

I'm not sure if my assertion is correct, as I can't estimate how accurately the log entries reflect the actual query. I tried to search with findByCursor and didn't get any hits. Then I found in the logs the expression and tried it with MongoDB Compass. 

According to [MongoDB Documentation](https://www.mongodb.com/docs/compass/current/query/filter/?utm_source=compass&utm_medium=product#match-by-date ) its necessary to wrap a date-attribute with new Date()
the change should be like:

before:
`{valuta: { '$lte': '2022-12-27T00:00:00.000Z' }}`

after
`{valuta: { $lte: new Date("2022-12-27T00:00:00.000Z") }}`

BTW: The log-entrie is also writing:
`'$or': [ [Object], [Object] ] `
I think there is something missing...

### Reproduction
Test-File:
[orderByWithDate.test.txt](https://github.com/mikro-orm/mikro-orm/files/15045774/orderByWithDate.test.txt)
(rename it to orderByWithDate.test.ts)

![image](https://github.com/mikro-orm/mikro-orm/assets/38237424/f0378d1b-e1be-438d-86d4-6f182db18a5c)

### What driver are you using?

@mikro-orm/mongodb

### MikroORM version

6.2.1

### Node.js version

v20.12.2

### Operating system

Ubuntu 22.04.4 LTS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Mongo docs are not really relevant when you use the ORM methods. Provide some code, logs are not a reproduction.

**Comment by lod911** [comment]
Hi @B4nan 

Found out, how it works with the tests. I updated this discussion with a test. Hope it gives the necessary point of the Problem.

Regards

**Reply by B4nan** [reply]
thanks, that's better, will look into this during the day

**Comment by lod911** [comment]
@B4nan 
Thank you.
Found another issue: The $like Operator doesn't work for MongoDB ([Documentation](https://www.mongodb.com/docs/compass/current/query/filter/?utm_source=compass&utm_medium=product#match-by-substring) )
It's $regex

`
      const [response, count] = await ctx.db.em.findAndCount(
        Transaction,
        { description: {$like: trFormated.description} },
        {
          populate: ['tags.*', 'bankAccount', 'bankAccount.bankAccountCompany'],
          limit: size,
          offset: offset,
        },
      );

`

gives:
    err: {
      "type": "DriverException",
      "message": "unknown operator: $like\nQuery: db.getCollection('transaction').countDocuments({ '$and': [ { tenantName: 'tn' }, { description: { '$like': [ 'Einkauf' ] } } ] }, {});",
      "stack":
          DriverException: unknown operator: $like
          Query: db.getCollection('transaction').countDocuments({ '$and': [ { tenantName: 'Mandant1' }, { description: { '$like': [ 'xy' ] } } ] }, {});

I will provide tomorrow a test and a new conversation. Just in case you currently working on it.

**Reply by B4nan** [reply]
`$like` operator is SQL only, as it literally represents the `like` SQL keyword.

**Comment by lod911** [comment]
OK, no problem. Did it with regexp:
`description: new RegExp(
            new String(value).replace(
              /[.*+?^${}()|[\]\\]/g,
              '\\$&',
            ),
            'i',
          ),`

**Comment by lod911** [comment]
@B4nan 
Did you found something for this orderBy behavior?
Regards

**Reply by B4nan** [reply]
This should help: 4281320f233efb56d012da4fe60d380e8aa701a3

**Comment by lod911** [comment]
Great. Thank you very much! Works perfectly!
